pal/src/exception/machmessage.h

```diff
@@ -55,6 +55,21 @@ typedef	mach_exception_data_type_t	mach_exception_subcode_t;
 
 struct MachExceptionHandler;
 
+class MachMessage;
+
+// Contains all the exception and thread state information needed to forward the exception.
```

Comment:
Can you update the PR description to explain the new set of changes around exception handling? I'm not too familiar with OS X exception handling to start out with and some more context about the motivation/some pointers to docs to read up on would be nice. Also, I see you have functions defined in the below struct but where are they implemented?